Dewikification - Display resources based on YAML files

ks129 opened the pull request:
Closes #388 

- Added view that display resources and sub-categories of resources.
- Icon colors is handled inside CSS.
- Migrated all YAML files to new format and made them up-to-date with current site.
- Pushed Django Simple Bulma version to latest to get tooltip extension.

ks129 committed:
Move resources files from home app to resources app

ks129 committed:
Update resources YAML files to match with new system

ks129 committed:
Create as_icon templatetag and tests for it

ks129 committed:
Push Django Simple Bulma version

ks129 committed:
Fix lint in resources home view

ks129 committed:
Create testing resources

ks129 committed:
Add Bulma Tooltip extension to settings

ks129 committed:
Create HTML files for resources lists

ks129 committed:
Create CSS for resources list

ks129 committed:
Create view and tests for this view for resources lists

ks129 committed:
Add resources lists view to __init__.py

ks129 committed:
Add URLs for resources lists

ks129 committed:
Use URL tags instead hard-coded strings for directing to specific list

ks129 committed:
Add Google Colab to editors

ks129 committed:
Add "Neural Networks from Scratch in Python" book to resources

ks129 committed:
Add "Python Crash Course" book to resources

ks129 committed:
Remove unnecessary leading newline from NNFSIP book

NNFSIP = Neural Networks from Scratch in Python

ks129 committed:
Add WTF Python tutorial resource

ks129 committed:
Add Inferential Thinking book to resources

ks129 committed:
Add JimShapedCoding YouTube channel to resources

ks129 committed:
Add accessibility tools category to tools resources

ks129 committed:
Add Talon Voice accessibility tool to resources

ks129 committed:
Add Screen Readers accessibility tools listing to resources

ks129 committed:
Add link to inferential thinking book resource

kosayoda reviewed (changes requested):
I'm getting errors at the following links:
- /resources/reading/
- /resources/videos/

Please check the YAML, since it's likely due to colons and quotes that aren't properly escaped/quoted in fields.

I'm also unsure about the payment info (pricing) being shown next to the resource entries, since we don't have it on the current site (although the information is in the original YAML).

Can @lemonsaurus advise on whether we want payment info?
![image](https://user-images.githubusercontent.com/41782385/112795053-8a9f6f80-909a-11eb-839c-b44a11f23fcd.png)
If we do want payment info, I suggest placing it under the description (like the other links in the `exercism` example. Also, it should have a tooltip explaining what each of the payment icons mean.

ks129 committed:
Remove everything payment-related from resources app

We decided not to include payment info right now.
Payment data will be handled in future.

ks129 committed:
Wrap some YAML resources descriptions to quotes to avoid parse errors

kosayoda commented on pydis_site/apps/resources/resources/reading/books/neural_networks_from_scratch_in_python.yaml:
```diff
@@ -1,8 +1,8 @@
   This is so you can go out and do new/novel things with deep learning as well as to become more successful with even more basic models.
   This book is to accompany the usual free tutorial videos and sample code from youtube.com/sentdex.
-name: Neural Networks from Scratch in Python
+name: Neural Networks from Scratch in Python'
```
You accidentally quoted into the `name` field, so the name isn't showing up now.

kosayoda commented on pydis_site/static/css/resources/resources_list.css:
```diff
@@ -0,0 +1,75 @@
+.payment-paid:hover {
+    color: goldenrod;
+}
+
```
Do we still need this?

kosayoda commented on pydis_site/apps/resources/resources/reading/tutorials/_category_info.yaml:
```diff
@@ -1,3 +1,6 @@
 description: Tutorials and references for those that are just getting started with
```
```suggestion
description: Tutorials and references for those that are just getting started with Python.
```

kosayoda commented on pydis_site/apps/resources/resources/reading/_category_info.yaml:
```diff
@@ -0,0 +1,2 @@
+description: Python related amazing books and guides.
```
```suggestion
description: Books related to Python and popular third-party libraries and frameworks.
```

kosayoda commented on pydis_site/apps/resources/resources/videos/_category_info.yaml:
```diff
@@ -0,0 +1,2 @@
+description: Best Youtubers who make videos about Python.
```
```suggestion
description: Excellent Youtube channels with content related to Python.
```

kosayoda commented on pydis_site/apps/resources/resources/interactive/_category_info.yaml:
```diff
@@ -1,3 +1,5 @@
 description: Learn Python with interactive content like courses, games and programming
   challenges.
```
```suggestion
description: Learn Python with interactive courses, games, and programming challenges.
```

kosayoda commented on pydis_site/apps/resources/resources/communities/subreddit.yaml:
```diff
@@ -0,0 +1,6 @@
+description: News about the Python programming language, and language-related discussion
```
```suggestion
description: News about the Python programming language, and language-related discussion.
```

kosayoda commented on pydis_site/apps/resources/resources/podcasts/_category_info.yaml:
```diff
@@ -1,2 +1,4 @@
 description: Notable podcasts about the Python ecosystem
```
```suggestion
description: Notable podcasts about the Python ecosystem.
```

kosayoda commented on pydis_site/apps/resources/resources/tools/ides/_category_info.yaml:
```diff
@@ -1,2 +1,5 @@
 description: Fully-integrated development environments for serious Python work
```
```suggestion
description: Fully-integrated development environments for serious Python work.
```

kosayoda commented on pydis_site/apps/resources/resources/tools/accessibility_tools/_category_info.yaml:
```diff
@@ -0,0 +1,5 @@
+description: Accessibility tools that helps people to write Python code
```
```suggestion
description: Accessibility tools that help people write Python code.
```

kosayoda commented on pydis_site/apps/resources/views/resources_list.py:
```diff
@@ -0,0 +1,58 @@
+                not resource_path.exists()
+                or not resource_path.is_dir()
+                or not resource_path.joinpath("_category_info.yaml").exists()
+        ):
```
`is_dir` returns `False` if the path does not exist

ks129 replied:
I think everything is correct here? (At least everything works locally)

kosayoda replied:
yesyes, `not resource_path.exists()` is redundant though

kosayoda commented on pydis_site/apps/resources/views/resources_list.py:
```diff
@@ -0,0 +1,58 @@
+        """Add resources and subcategories data into context."""
+        context = super().get_context_data(**kwargs)
+
+        resource_path = RESOURCES_PATH / self.kwargs["type"]
```
Since we refer to it everywhere else as `category`, shouldn't this be `category` rather than `type`?

kosayoda commented on pydis_site/settings.py:
```diff
@@ -273,5 +273,7 @@
-    }
+        "tooltip-max-width": "30rem",
+    },
+    "extensions": ["bulma-tooltip"],
```
Is this extension used anywhere?

kosayoda commented on pydis_site/apps/resources/views/resources_list.py:
```diff
@@ -0,0 +1,58 @@
+        )
+        context["category_info"] = {**yaml.safe_load(
+            resource_path.joinpath("_category_info.yaml").read_text()
+        ), "raw_name": resource_path.name}
```
Can this be refactored into helper methods that get category and resource information? Secondly, Django has the [`dictsort` filter](https://docs.djangoproject.com/en/3.1/ref/templates/builtins/#dictsort), so we don't have to sort in the Python code.

kosayoda commented on pydis_site/apps/resources/views/resources_list.py:
```diff
@@ -0,0 +1,58 @@
+RESOURCES_PATH = Path(settings.BASE_DIR, "pydis_site", "apps", "resources", "resources")
+
+
+class ResourcesList(TemplateView):
```
This should be `ResourcesListView`.

kosayoda reviewed (commented):
Works well, nice job!

Suggestions for files where I can't act using the Github UI:
- File: `resources/communities/_category_info.yaml `
- Description: `Partnered communities that share part of our mission.`

ks129 committed:
Rename ResourcesList -> ResourcesListView

ks129 committed:
Remove unnecessary exists check from resources path check

ks129 committed:
Refactor getting subcategories and resources to 2 helper methods

ks129 committed:
Order resources and subcategories in template

ks129 committed:
Remove unused Bulma Tooltip extension

ks129 committed:
Rename resource type -> resource category in route

ks129 committed:
Improve resources categories descriptions

ks129 committed:
Remove resources payments-related CSS

ks129 committed:
Add missing dot to subreddit YAML description

ks129 committed:
Move quotes ending to right place so this don't take name as description

ks129 committed:
Fix applying description changes to wrong places

ks129 committed:
Merge branch 'dewikification' into resources-lists

ks129 committed:
Relock Pipfile

kosayoda commented on pydis_site/apps/resources/resources/reading/books/_category_info.yaml:
```diff
@@ -1,2 +1,5 @@
 description: The best books for learning Python or Python Frameworks
```
```suggestion
description: The best books for learning Python or Python Frameworks.
```

kosayoda commented on pydis_site/apps/resources/resources/podcasts/the_real_python_podcast.yaml:
```diff
@@ -0,0 +1,7 @@
+  career tips, and related software development topics.
+name: The Real Python Podcast
+title_url: https://realpython.com/podcasts/rpp/
+postion: 4
```
```suggestion
position: 4
```

ks129 committed:
Add missing dot to end of books category info

Co-authored-by: Kieran Siek <kieransiek@protonmail.com>

ks129 committed:
Change typo postion -> position in Real Python podcast resource

Co-authored-by: Kieran Siek <kieransiek@protonmail.com>
